Add algebraic reinitialisation to theory guide (#1459)
chaos-polymtl/lethe: doc/source/theory/multiphase/cfd/vof.rst

> The VOF method tends to diffuse the interface, i.e., over time, the interface becomes blurry instead of a sharp definition, and the change from :math:`\phi = 0` to :math:`1` happens on a larger distance.

Reviewer: ```suggestion
The VOF method tends to diffuse the interface, i.e., over time, the interface becomes blurry instead of a sharp definition, and the change from :math:`\phi = 0` to :math:`1` occurs on a larger distance.
```

a phase fraction threshold (generally :math:`0.5`), and :math:`\alpha` corresponds to the interface sharpness, which is a model parameter generally in the range of :math:`(1,2]`. This interface sharpening method was proposed by Aliabadi and Tezduyar (2000) [#aliabadi2000]_.

""""""""""""""""""""""""""""""""""""""""
Algebraic Interface Reinitialization
""""""""""""""""""""""""""""""""""""""""

> The algebraic interface reinitialization method consists of compressing and diffusing the interface in the normal direction of the interface. It is done by solving the following transient partial differential equation (PDE) until steady-state is reached using a pseudo-time-stepping scheme as proposed by Olsson and coworkers (2007) [#olsson2007]_:

Reviewer: ```suggestion
The algebraic interface reinitialization method consists of compressing and diffusing the interface in its normal direction. This is done by solving the following transient partial differential equation (PDE) until steady-state is reached using a pseudo-time-stepping scheme as proposed by Olsson and coworkers (2007) [#olsson2007]_:
```

.. math::

    \underbrace{\frac{\partial\phi_\text{reinit}}{\partial \tau}}_\text{transient}
     + \underbrace{\nabla \cdot \left[ \phi_\text{reinit} (1-\phi_\text{reinit})\mathbf{n}\right]}_\text{compression}
    = \underbrace{\varepsilon \nabla \cdot \left[  (\nabla \phi_\text{reinit} \cdot \mathbf{n}) \mathbf{n}\right]}_\text{diffusion}

where:

- :math:`\phi_\text{reinit}` is the reinitialized phase fraction;

> - :math:`\tau` is the pseudo-time variable. It is different from the time variable :math:`t` of the actual simulation.

Reviewer: ```suggestion
- :math:`\tau` is the pseudo-time independent variable. It is different from the time independent variable :math:`t` of the actual simulation.
```
Just to highlight that it's a time, like x is a position and etc.

- :math:`\mathbf{n} = \frac{\nabla \psi}{\lVert \nabla \psi \rVert_2}` is the normal vector of the interface with :math:`\nabla \psi` the projected VOF phase gradient, and;

> - :math:`\varepsilon = C \, h_\text{min}^d` is the diffusion coefficient with  :math:`h_\text{min}` the smallest cell-size, :math:`C` a constant factor multiplying :math:`h_\text{min}`, and :math:`d` a constant power to which :math:`h_\text{min}` is elevated.

Reviewer 2: Can you give typical values fore C and d?

As the equation is non-linear, we use the `Newton-Raphson method <https://en.wikipedia.org/wiki/Newton%27s_method>`_ and solve the linear system :math:`\mathcal{J} \, \delta\phi_\text{reinit} = -\mathcal{R}`. We obtain the Residual (:math:`\mathcal{R}`) and Jacobian (:math:`\mathcal{J}`) below:

.. math::
    \begin{split}
    \mathcal{R} = & \int_\Omega \upsilon \, \partial_\tau \phi_\text{reinit}^{n} \, \mathrm{d}\Omega \\
                   & + \int_\Omega\upsilon
                  \left[ \mathbf{n} \cdot \nabla \phi_\text{reinit}^{n} - \mathbf{n} \cdot \left(  2 \phi_\text{reinit}^{n} \nabla \phi_\text{reinit}^{n}\right) +(\phi_\text{reinit}^{n} -{\left(\phi_\text{reinit}^{n}\right)}^2) (\nabla \cdot \mathbf{n}) \right] \mathrm{d}\Omega \\
                   & + \int_\Omega \nabla\upsilon \cdot \varepsilon (\nabla \phi_\text{reinit}^{n} \cdot \mathbf{n}) \mathbf{n} \, \mathrm{d} \Omega

    \end{split}

> where, :math:`\phi_\text{reinit}^{n} = \phi_\text{reinit}^{n-1} + \delta \phi_\text{reinit}` is the reinitialized phase fraction value of the previous Newton iteration updated following .

Reviewer 2: ```suggestion
where, :math:`\phi_\text{reinit}^{n}` is the reinitialized phase fraction value of the previous Newton iteration.
```
